File: tasks/2025/round4_pp/valentinesday-pp.py
#!/usr/bin/env python3
# @check-accepted: *

N, M = map(int, input().split())
C = list(map(int, input().split()))
P = list(map(int, input().split()))

maxChocolates = 0

for i in range(N):
    if C[i] % 2 == 0:
        if P[i] <= M:
            maxChocolates = max(maxChocolates, C[i])

print(maxChocolates)

File: tasks/2025/round4_pp/valentinesday-pp.cpp
// @check-accepted: *

#include <iostream>
#include <vector>

using namespace std;

int main() {
  int N, M;
  cin >> N >> M;

  vector<int> C(N);
  for (int i = 0; i < N; ++i) {
    cin >> C[i];
  }
  
  vector<int> P(N);
  for (int i = 0; i < N; ++i) {
    cin >> P[i];
  }
  
  int maxChocolates = 0;
  
  for (int i = 0; i < N; ++i) {
    if (C[i] % 2 == 0 && P[i] <= M && C[i] > maxChocolates) {
      maxChocolates = C[i];
    }
  }
  
  cout << maxChocolates << endl;
  
  return 0;
}
